Add custom wrench "disassembly" mechanics for certain blocks.

eutro opened the pull request:
This PR reimplements the disassembly of Creative Tanks using CraftTweaker, with the intention of allowing non-OP players to do it too. This means that server players aren't handicapped in the post-tank stage.

Changes:

- Non-OPs can disassemble Creative Tanks
- More wrenches can be used, not just the Crescent Hammer:
  - Yeta Wrench
  - Flux-infused Wrench
  - Flux-infused Battlewrench
  - Gregtech Wrenches

Commit:
Attempt to simulate Creative Tank disassembly.

Commit:
Move creative tank wrenching to a separate file, extract condition functions, cancel event.

eutro committed:
Fix Creative Tank wrenching checks.

eutro committed:
Write the correct NBT to the tank.

eutro committed:
Remove #debug

eutro committed:
Transfer name and holding enchantment.

eutro committed:
Transfer security.
- Appears to cause desync when the tile is secured.

eutro committed:
Summon entity directly, fixes desync.

eutro committed:
Support more wrenches.

fabrossmann commented:
Hey we tried it on our Server and it is not working correctly. If you try to remove it, the tank stays there and you will get 3 extra.
![Alt Text](https://i.imgur.com/M4NMrrO.gif)

We tried it in Version 1.2.2.

eutropius225 commented:
It being a SpongeForge server could definitely cause issues. What are all the changes that have been made to the pack?

fabrossmann commented:
We run Exa's Fixes and have some SpongeForge Mods/Plugins running. We also have 2 CT Scripts for custom recipes.
I also tried it now on a Omnifactory Sponge Server without any extra Mods or Scripts and getting this null error:
![image](https://user-images.githubusercontent.com/6189957/90554488-3fcd8900-e196-11ea-84fb-bfa213a5316e.png)
In singleplayer we get this error:
![image](https://user-images.githubusercontent.com/6189957/90554065-a1412800-e195-11ea-8e3c-69141f4e06c4.png)

eutropius225 commented:
Oh that's right, the script wouldn't work without MultiblockTweaker. I will change that now.

eutro committed:
Check for removal success before dropping.
- Should remove any possibility of duping.
- Use CrT's own BlockState bracket handler.

eutro committed:
Update CraftTweaker, use getPickedBlock

eutro committed:
Add more wrenchable blocks.

ALongStringOfNumbers commented:
Having issues with the disassembly working on the Creative Tank with a damaged GregTech (Darmstadtium) Wrench. In addition with the Creative Tank, left clicking with the GTCE wrench does not change the output side of the tank, while all the other wrenches do change the output side. Personally, I think the GTCE wrench should be not included due to the issues that might arise from getting the wrench to work correctly, and also it is a known factor that the GTCE wrench really only works on GTCE machines, and not much intermod. After a little more testing, the damaged GTCE wrench worked for none of the items added to the wrenching list

eutro committed:
Remove GregTech wrench from the wrenchCustom oredict.

eutro committed:
Wrename script.
